packages/dotnet/src/lib/core/dotnet.client.ts

```diff
@@ -24,160 +23,161 @@ import { LoadedCLI } from './dotnet.factory';
     extraParameters?: string,
-  ): void {
-    let cmd = `${this.cliCommand.command} publish "${project}"`;
+  ): Buffer {
+    const params = [`publish`, `"${project}"`];
     if (parameters) {
       parameters = swapKeysUsingMap(parameters, publishKeyMap);
-      const paramString = parameters ? getParameterString(parameters) : '';
-      cmd = `${cmd} ${paramString}`;
+      params.push(...getParameterArrayStrings(parameters));
     }
     if (publishProfile) {
-      cmd = `${cmd} -p:PublishProfile=${publishProfile}`;
+      params.push(`-p:PublishProfile=${publishProfile}`);
     }
     if (extraParameters) {
-      cmd = `${cmd} ${extraParameters}`;
+      params.push(`${extraParameters}`);
     }
-    return this.logAndExecute(cmd);
+    return this.logAndExecute(params);
   }
 
-  installTool(tool: string): void {
-    const cmd = `${this.cliCommand.command} tool install ${tool}`;
+  installTool(tool: string): Buffer {
+    const cmd = [`tool`, `install`, tool];
     return this.logAndExecute(cmd);
   }
 
-  restorePackages(project: string): void {
-    const cmd = `${this.cliCommand.command} restore "${project}"`;
+  restorePackages(project: string): Buffer {
+    const cmd = [`restore`, project];
     return this.logAndExecute(cmd);
   }
 
-  restoreTools(): void {
-    const cmd = `${this.cliCommand.command} tool restore`;
+  restoreTools(): Buffer {
+    const cmd = [`tool`, `restore`];
     return this.logAndExecute(cmd);
   }
 
-  format(project: string, parameters?: dotnetFormatOptions): void {
-    let cmd = `${this.cliCommand.command} format "${project}"`;
+  format(project: string, parameters?: dotnetFormatOptions): Buffer {
+    const params = [`format`, project];
     if (parameters) {
       parameters = swapKeysUsingMap(parameters, formatKeyMap);
-      const paramString = parameters ? getParameterString(parameters) : '';
-      cmd = `${cmd} ${paramString}`;
+      params.push(...getParameterArrayStrings(parameters));
     }
-    return this.logAndExecute(cmd);
+    return this.logAndExecute(params);
   }
 
   addProjectToSolution(solutionFile: string, project: string) {
-    const cmd = `${this.cliCommand.command} sln "${solutionFile}" add "${project}"`;
-    this.logAndExecute(cmd);
+    const params = [`sln`, solutionFile, `add`, project];
+    this.logAndExecute(params);
   }
 
   getSdkVersion(): Buffer {
-    const cmd = 'dotnet --version';
-    return this.execute(cmd);
+    return this.execute(['--version']);
   }
 
   printSdkVersion(): void {
-    this.logAndExecute('dotnet --version');
+    this.logAndExecute(['--version']);
   }
 
-  private logAndExecute(cmd: string): void {
-    console.log(`Executing Command: ${cmd}`);
-    execSync(cmd, { stdio: 'inherit', cwd: this.cwd || process.cwd() });
+  private logAndExecute(params: string[]): Buffer {
+    console.log(
+      `Executing Command: ${this.cliCommand.command} ${params.join(', ')}`,
+    );
+    return this.execute(params);
```

Comment:
This method needs the 'stdio: inherit' so that things like `nx build` show terminal output.